test/js/ui/map.test.js

```diff
@@ -883,26 +883,34 @@ test('Map', function(t) {
 
     t.test('error event', function (t) {
         t.test('logs errors to console when it has NO listeners', function (t) {
+            var map = createMap({ style: { version: 8, sources: {}, layers: [] } });
+
+            sinon.spy(map, 'fire');
             sinon.stub(console, 'error', function(error) {
-                console.error.restore();
-                t.equal(error.message, 'version: expected one of [8], 7 found');
-                t.end();
+                if (error.message === 'version: expected one of [8], 7 found') {
+                    t.notOk(map.fire.calledWith('error'));
+                    console.error.restore();
+                    map.fire.restore();
+                    t.end();
+                } else {
+                    console.log(error);
```

Comment:
Modifying the global `console.error` method is tricky. This test was failing due to errors from other tests' map instances being emitted asynchronously after their test had completed. In this case, we don't want this test to fail and we do want the error message printed to the console (either the error message contains useful information or it shouldn't have been emitted in the first place).